backend/clickhouse/traces.go

```diff
@@ -122,63 +122,105 @@ var TracesSampleableTableConfig = SampleableTableConfig{
-	SecureSessionId  string
-	TraceState       string
-	SpanName         string
-	SpanKind         string
-	Duration         int64
-	ServiceName      string
-	ServiceVersion   string
-	TraceAttributes  map[string]string
-	StatusCode       string
-	StatusMessage    string
-	Environment      string
-	HasErrors        bool
-	EventsTimestamp  []time.Time         `json:"Events.Timestamp" ch:"Events.Timestamp"`
-	EventsName       []string            `json:"Events.Name" ch:"Events.Name"`
-	EventsAttributes []map[string]string `json:"Events.Attributes" ch:"Events.Attributes"`
-	LinksTraceId     []string            `json:"Links.TraceId" ch:"Links.TraceId"`
-	LinksSpanId      []string            `json:"Links.SpanId" ch:"Links.SpanId"`
-	LinksTraceState  []string            `json:"Links.TraceState" ch:"Links.TraceState"`
-	LinksAttributes  []map[string]string `json:"Links.Attributes" ch:"Links.Attributes"`
+	Timestamp           time.Time
+	UUID                string
+	TraceId             string
+	SpanId              string
+	ParentSpanId        string
+	ProjectId           uint32
+	SecureSessionId     string
+	TraceState          string
+	SpanName            string
+	SpanKind            string
+	Duration            int64
+	ServiceName         string
+	ServiceVersion      string
+	TraceAttributes     map[string]string
+	StatusCode          string
+	StatusMessage       string
+	Environment         string
+	HasErrors           bool
+	EventsTimestamp     []time.Time         `json:"Events.Timestamp" ch:"Events.Timestamp"`
+	EventsName          []string            `json:"Events.Name" ch:"Events.Name"`
+	EventsAttributes    []map[string]string `json:"Events.Attributes" ch:"Events.Attributes"`
+	LinksTraceId        []string            `json:"Links.TraceId" ch:"Links.TraceId"`
+	LinksSpanId         []string            `json:"Links.SpanId" ch:"Links.SpanId"`
+	LinksTraceState     []string            `json:"Links.TraceState" ch:"Links.TraceState"`
+	LinksAttributes     []map[string]string `json:"Links.Attributes" ch:"Links.Attributes"`
+	HttpResponseBody    string
+	HttpRequestBody     string
+	HttpUrl             string
+	HighlightKey        string
+	HighlightType       string
+	HttpAttributes      map[string]string
+	ProcessAttributes   map[string]string
+	OsAttributes        map[string]string
+	TelemetryAttributes map[string]string
+	WsAttributes        map[string]string
+	EventAttributes     map[string]string
+	DbAttributes        map[string]string
+}
+
+func getSubAttributes(attributes map[string]string, prefix string) map[string]string {
+	results := map[string]string{}
+	for k, v := range attributes {
+		if strings.HasPrefix(k, prefix) {
+			results[k] = v
+		}
+	}
+	return results
+}
+
+func getHttpAttributes(attributes map[string]string) map[string]string {
+	httpAttributes := getSubAttributes(attributes, "http.")
+	delete(httpAttributes, "http.response.body")
+	delete(httpAttributes, "http.request.body")
+	delete(httpAttributes, "http.url")
```

Comment:
makes sense, this is removing these keys from `TraceAttributes` since they will now be on the dedicated column. shouldn't we do this in `getSubAttributes` for all of the attributes we bring out? or are we just starting with these because these are the largest keys (so they will  have the most impact on being removed from TraceAttributes)

Reply:
the `delete` here is just removing them from `httpAttributes` and not `TraceAttributes` - I'll be removing all of these from `TraceAttributes` as the last step in this migration but for now we need to dual write since we'll still be referencing `TraceAttributes` until these new columns are backfilled and new code to query the new columns is deployed

Reply:
and I'm giving these 3 their own column because they're like >70% of all ingested data. a little hacky but should have a large impact for queries that aren't referencing the request/response body

Reply:
> I'll be removing all of these from TraceAttributes as the last step in this migration but for now we need to dual write

ahh makes sense, ty for explainng